tools/shared.py

```diff
@@ -809,7 +809,7 @@ def get_subresource_location(path, data_uri=None):
         return ''
       with open(path, 'rb') as f:
         data = base64.b64encode(f.read())
-      return 'data:application/octet-stream;base64,' + asstr(data)
+      return 'data:application/octet-stream;base64,' + data.decode('ascii')
```

Comment:
why is this ascii and not utf-8? that may make this NFC.

Reply:
Oh yes, I did little research about why `b64encode` does't return a string. 

The discussion is here: https://stackoverflow.com/questions/42773386/why-does-base64-b64encode-return-a-bytes-object

The recommended answer is just to use `.decode('ascii')`.. since data is the output b64encode there is no way it contains non-ascii chars.. and it if did we would rather crash here.. but it can't.

Reply:
I see, thanks, sgtm.